controllers/sample_controller_rendered_resources.go

```diff
@@ -0,0 +1,359 @@
+		return ctrl.Result{RequeueAfter: requeueInterval}, r.HandleReadyState(ctx, &objectInstance)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// HandleInitialState bootstraps state handling for the reconciled resource.
+func (r *SampleReconciler) HandleInitialState(ctx context.Context, objectInstance *v1alpha1.Sample) error {
+	status := getStatusFromObjectInstance(objectInstance)
+
+	installationCondition := metav1.Condition{
+		Type:               ConditionTypeInstallation,
+		Reason:             ConditionReasonReady,
+		Status:             metav1.ConditionFalse,
+		Message:            "installation is ready and resources can be used",
+		ObservedGeneration: objectInstance.GetGeneration(),
+	}
+
+	status.Conditions = make([]metav1.Condition, 0, 1)
+	meta.SetStatusCondition(&status.Conditions, installationCondition)
+
+	return r.setStatusForObjectInstance(ctx, objectInstance, status.WithState(v1alpha1.StateProcessing))
+}
+
+// HandleProcessingState processes the reconciled resource by processing the underlying resources.
+// Based on the processing either a success or failure state is set on the reconciled resource.
+func (r *SampleReconciler) HandleProcessingState(ctx context.Context, objectInstance *v1alpha1.Sample) error {
+	r.Event(objectInstance, "Warning", "Processing", "resource processing")
+	logger := log.FromContext(ctx)
+
+	status := getStatusFromObjectInstance(objectInstance)
+
+	resourceObjs, err := getResourcesFromLocalPath(objectInstance.Spec.ResourceFilePath, logger)
+	if err != nil {
+		logger.Error(err, "error locating manifest of resources")
+		r.Event(objectInstance, "Warning", "ResourcesInstall", "locating resources error")
+		return r.setStatusForObjectInstance(ctx, objectInstance, status.WithState(v1alpha1.StateError))
+	}
+
+	// the resources to be installed are unstructured,
+	// so please make sure the types are available on the target cluster
+	for _, obj := range resourceObjs.Items {
+		r.Event(objectInstance, "Normal", "ResourcesInstall", "installing resources")
+		if err = r.ssa(ctx, obj); err != nil && !errors2.IsAlreadyExists(err) {
+			logger.Error(err, "error during installation of resources")
+			r.Event(objectInstance, "Warning", "ResourcesInstall", "installing resources error")
+			return r.setStatusForObjectInstance(ctx, objectInstance, status.WithState(v1alpha1.StateError))
+		}
+	}
+
+	status.Conditions[0].Status = metav1.ConditionTrue
+	status.Conditions[0].ObservedGeneration = objectInstance.GetGeneration()
+	meta.SetStatusCondition(&status.Conditions, status.Conditions[0])
+	return r.setStatusForObjectInstance(ctx, objectInstance, status.WithState(v1alpha1.StateReady))
+}
+
+// HandleDeletingState processed the deletion on the reconciled resource.
+// Once the deletion if processed the relevant finalizers (if applied) are removed.
+func (r *SampleReconciler) HandleDeletingState(ctx context.Context, objectInstance *v1alpha1.Sample) error {
+	r.Event(objectInstance, "Warning", "Deleting", "resource deleting")
```

Comment:
why do you issue a warning if the resource is getting deleted? isn't that normal behaviour?

Reply:
copy paste error :)